### PR TITLE
[ozone/headless] Add HeadlessNativeDisplayDelegate.

### PR DESCRIPTION
**Pull request by jkim-julie**
This CL created HeadlessNativeDisplayDelegate for ozone/headless.
DesktopScreenOzone keeps it at delegate_ and communicate it.

With thie CL, we can launch chromium with '--ozone-platform=headless'.

### CODE REVIEW

**Comment by jkim-julie** [comment]
@msisov, HeadlessNativeDisplayDelegate is created. When I checked
headless without ozone, window size is set with defalut value at
https://cs.chromium.org/chromium/src/headless/public/headless_browser.cc?rcl=28b6d51b55308bc3d8562414859640161d0d3e93&l=50
and can be set with command line argument,
at https://cs.chromium.org/chromium/src/headless/public/headless_browser.cc?rcl=28b6d51b55308bc3d8562414859640161d0d3e93&l=160

And then HeadlessScreen is created at https://cs.chromium.org/chromium/src/headless/lib/browser/headless_browser_impl_aura.cc?rcl=28b6d51b55308bc3d8562414859640161d0d3e93&l=23
and it calls SetScreenInstance().

This CL doesn't include the part for respecting argument window size. I need to consider how
to make it.

PTAL.

File: ui/ozone/platform/headless/headless_native_display_delegate.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  bool SetGammaCorrection(
+      int64_t display_id,
+      const std::vector<display::GammaRampRGBEntry>& degamma_lut,
+      const std::vector<display::GammaRampRGBEntry>& gamma_lut) override;
+  void AddObserver(display::NativeDisplayObserver* observer) override;
+  void RemoveObserver(display::NativeDisplayObserver* observer) override;
+  display::FakeDisplayController* GetFakeDisplayController() override;
+
+ private:
+  int64_t get_next_display_id() { return next_display_id_++; }
```
Comment (msisov) [inline ui/ozone/platform/headless/headless_native_display_delegate.h:46]: nit: space after get_next_display_id()

File: ui/ozone/platform/headless/headless_native_display_delegate.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  std::unique_ptr<display::DisplaySnapshot> current_snapshot_;
+  std::unique_ptr<display::DisplayMode> current_mode_;
+
+  base::ObserverList<display::NativeDisplayObserver> observers_;
+  // The next available display id.
+  uint8_t next_display_id_ = 0;
```
Comment (msisov) [inline ui/ozone/platform/headless/headless_native_display_delegate.h]: s/uint8_t/uint64_t/

File: ui/ozone/platform/headless/headless_native_display_delegate.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  void AddObserver(display::NativeDisplayObserver* observer) override;
+  void RemoveObserver(display::NativeDisplayObserver* observer) override;
+  display::FakeDisplayController* GetFakeDisplayController() override;
+
+ private:
+  int64_t get_next_display_id() { return next_display_id_++; }
+  std::unique_ptr<display::DisplaySnapshot> current_snapshot_;
+  std::unique_ptr<display::DisplayMode> current_mode_;
+
+  base::ObserverList<display::NativeDisplayObserver> observers_;
```
Comment (msisov) [inline ui/ozone/platform/headless/headless_native_display_delegate.h:50]: space after.

File: ui/ozone/platform/headless/headless_native_display_delegate.cc [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+HeadlessNativeDisplayDelegate::HeadlessNativeDisplayDelegate() = default;
+
+HeadlessNativeDisplayDelegate::~HeadlessNativeDisplayDelegate() = default;
+
+void HeadlessNativeDisplayDelegate::Initialize() {
+  // This shouldn't be called twice.
+  DCHECK(!current_snapshot_);
+
+  if (next_display_id_ == 0xFF) {
+    LOG(ERROR) << "Exceeded display id limit";
```
Comment (msisov) [inline ui/ozone/platform/headless/headless_native_display_delegate.cc]: LOG(FATAL)?

Comment (jkim-julie) [reply ui/ozone/platform/headless/headless_native_display_delegate.cc]: Done.

File: ui/ozone/platform/headless/headless_native_display_delegate.h (hunk 1 above)
Comment (jkim-julie) [reply ui/ozone/platform/headless/headless_native_display_delegate.h:46]: There is a space after get_next_display_id(). Could you check it again?

File: ui/ozone/platform/headless/headless_native_display_delegate.h (hunk 2 above)
Comment (jkim-julie) [reply ui/ozone/platform/headless/headless_native_display_delegate.h]: Done.

File: ui/ozone/platform/headless/headless_native_display_delegate.h (hunk 3 above)
Comment (jkim-julie) [reply ui/ozone/platform/headless/headless_native_display_delegate.h:50]: Done.

**Comment by jkim-julie** [comment]
@msisov , the branch was updated. PTAL.

File: ui/ozone/platform/headless/headless_native_display_delegate.cc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  current_snapshot_.reset(new display::DisplaySnapshot(
+      get_next_display_id(), gfx::Point(0, 0), kDefaultWindowSize,
+      display::DisplayConnectionType::DISPLAY_CONNECTION_TYPE_NONE, false,
+      false, false, gfx::ColorSpace(), "", base::FilePath(),
+      display::DisplaySnapshot::DisplayModeList(), std::vector<uint8_t>(),
+      nullptr, nullptr, 0, 0, gfx::Size()));
+
+  const int default_refresh = 60;
```
Comment (msisov) [inline ui/ozone/platform/headless/headless_native_display_delegate.cc]: move to the unnamed namespace and make it constexpr uint.

**Review by msisov — APPROVED** [review]
lgtm with one nit

**Comment by jkim-julie** [comment]
Updated branch and fixed nits.

**Pull request merged**